### ISSUE TITLE
Detecting an specific group of people on an image

### ISSUE THREAD

**Issue by SebastianJanampa** [ultralytics/ultralytics#18100]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello,
I am new to using the Ultralytics library. I am working on a project that involves detecting people close to the camera. The approach I want to follow is to use an RGB image + its depth map (estimated by another model such as DepthAnythingV2) as input. I want to use YOLOv11 but need help figuring out where to start. 

First, the YOLOv11 detection model detects multiple objects (80 classes if I remember correctly), but I only need to detect humans. So, my first question is how I can modify YOLOv11 to detect only humans but keep the pre-training weights from COCO because I don't have a large dataset.

Second, how can I concatenate the RGB image with the depth map? If I were using only pytorch, the solution would be simple. I just had to upload the RGB image and the depth map and use the torch cat function. However, ultralytics has a compile function for all his models, model.train(). 

Third, this is about YOLO's final predictions. Please correct me if I am wrong since I am learning about YOLO. YOLO uses anchors to predict the class and a bounding box for each detected object, followed by non-maximum suppression to remove overlapping objects. I want to know if I can retrieve the final anchors (the ones selected after NMS) to make them interact with the depth map; that way, I can freeze the human detection model and only the part of the model that distinguishes which person is close to the camera. Basically, I want to replicate what mask DINO does but use the anchors instead of queries.   

Fourth, is there any way to train YOLOv11 from scratch using the COCO dataset but only to detect human objects?

I really appreciate any help you can provide.
Sebastian :)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @SebastianJanampa, thank you for your interest in Ultralytics 🚀! We appreciate you taking the time to provide detailed information about your use case.

We recommend exploring our [Docs](https://docs.ultralytics.com) to get started with Ultralytics YOLOv8, as it covers topics ranging from [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage to customizing your training workflow and adapting models for your specific needs.

To address your questions:

1. If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/), as it helps us thoroughly understand and debug the issue.

2. If this is a ❓ Question about custom training or specific workflows, please share as much detail as possible about your setup, such as:
   - Dataset structure and examples (if applicable)
   - Any code you've attempted
   - Training or model details like input configuration and logs
   - Your current project goals

3. Regarding YOLO's compatibility with depth maps and modifying target classes, these could involve more advanced customizations, and your considerations about modifying the architecture or interacting with predictions are great starting points. An Ultralytics engineer will follow up soon to assist with these technical details! 😊

Beyond this, don't forget to join our community to connect and discuss with developers working on similar projects:
- 🎧 [Discord](https://discord.com/invite/ultralytics) for real-time chat
- 🗨️ [Discourse](https://community.ultralytics.com) for in-depth discussions
- 📚 [Subreddit](https://reddit.com/r/Ultralytics) to share and exchange insights

---

## Upgrade Instructions

Ensure you're running the latest version of `ultralytics` and dependencies to avoid potential compatibility issues related to your questions. You can upgrade as follows:

```bash
pip install -U ultralytics
```

You can also check all dependencies in the [requirements file](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml).

---

## Verified Environments

To simplify experiments, we recommend running YOLO in one of the following verified environments:
- **Notebooks** with GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/), [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/) or [Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) environments are also supported.

---

## Status Check

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [CI tests](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) are passing, validating YOLO's functionality on macOS, Windows, and Ubuntu.

Let us know if you encounter any further issues, and stay tuned—an Ultralytics engineer will be here to assist you shortly! 🚀

**Comment by glenn-jocher** [comment]
@SebastianJanampa hi Sebastian, great questions!

1. To detect only humans with YOLO11, utilize the `classes` argument during inference. For example: `model.predict(source='image.jpg', classes=[0])` where COCO class `[0]` corresponds to humans. For training, adjust your dataset YAML file to include only the "person" class and fine-tune a COCO-pretrained model (`yolo11n.pt`).

2. To concatenate RGB and depth maps, you will need to preprocess these inputs before passing them to YOLO11. Currently, YOLO11 models do not natively support multi-channel non-RGB inputs. You may modify the YOLO model architecture in PyTorch by extending its input channels and incorporating depth information into training.

3. YOLO does use anchors and applies NMS for overlapping detection. To interact with final bounding boxes, extract them directly from the prediction results using `results[0].boxes`. These boxes contain spatial information that you can combine with depth maps for your desired application.

4. For COCO training restricted to humans, filter COCO annotations for the "person" class, generate a custom dataset, and train with YOLO11. This effectively creates a human-specific detector from scratch leveraging COCO's subset.

For further details, check the [training guide](https://docs.ultralytics.com/modes/train/) or explore model customization options directly in the repository.

**Comment by Y-T-G** [comment]
> So, my first question is how I can modify YOLOv11 to detect only humans but keep the pre-training weights from COCO because I don't have a large dataset.

Weights are not class-specific. They can't be modified to remove other classes. You will have to filter the extra classes in the post-processing. You can pass `classes=[0]` to `model.predict()` to achieve that.

> Second, how can I concatenate the RGB image with the depth map? If I were using only pytorch, the solution would be simple. I just had to upload the RGB image and the depth map and use the torch cat function. However, ultralytics has a compile function for all his models, model.train().

Ultralytics only supports training with 3 channels. You need to make several modifications to change the number of channels since all the augmentations are based on the expectation that the inputs are 3 channel images.

> Third, this is about YOLO's final predictions. Please correct me if I am wrong since I am learning about YOLO. YOLO uses anchors to predict the class and a bounding box for each detected object, followed by non-maximum suppression to remove overlapping objects. I want to know if I can retrieve the final anchors (the ones selected after NMS) to make them interact with the depth map; that way, I can freeze the human detection model and only the part of the model that distinguishes which person is close to the camera. Basically, I want to replicate what mask DINO does but use the anchors instead of queries.

You can get the NMS indices by modifying the NMS function.

You can see a modified version that does that [here](https://y-t-g.github.io/tutorials/yolo-object-features/).


> Fourth, is there any way to train YOLOv11 from scratch using the COCO dataset but only to detect human objects

You will have to remove those extra labels from the COCO dataset and train.

**Comment by SebastianJanampa** [comment]
Thank you, @glenn-jocher  and @Y-T-G, for your valuable comments. I will try to implement them.

**Comment by glenn-jocher** [comment]
You're welcome, @SebastianJanampa! Best of luck with your implementation. Feel free to reach out if you encounter further issues or have additional questions related to Ultralytics YOLO.
